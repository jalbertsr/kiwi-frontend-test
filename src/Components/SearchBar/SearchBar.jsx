import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Input } from 'antd'
import styles from './SearchBar.css'

export default class SearchBar extends Component {
  constructor (props) {
    super(props)

    this.handleChange = this.handleChange.bind(this)
  }

  handleChange (e) {
    this.props.onChange({[this.props.use]: e.target.value})
  }

  render () {
    return (
      <Input onChange={this.handleChange} className={styles.input} placeholder={this.props.placeholder} />
    )
  }
}

SearchBar.propTypes = {
  onChange: PropTypes.func
}
