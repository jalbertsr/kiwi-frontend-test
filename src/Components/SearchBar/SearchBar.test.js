import SearchBar from './SearchBar'
import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'

configure({ adapter: new Adapter() })

/* eslint-disable no-unused-vars */

describe('SearchBar', () => {
  const jestMock = jest.fn()
  const props = {
    use: 'from',
    placeholder: 'Brno',
    onChange: jestMock
  }
  test('renders correctly', () => {
    const component = shallow(<SearchBar {...props} />)
    expect(component).toMatchSnapshot()
  })
})
