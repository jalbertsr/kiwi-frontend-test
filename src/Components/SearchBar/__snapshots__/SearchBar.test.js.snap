// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    onChange={[Function]}
    placeholder="Brno"
    use="from"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "choiceTransitionName": "zoom",
      "className": "input",
      "dataSource": Array [],
      "filterOption": false,
      "onChange": [Function],
      "optionLabelProp": "children",
      "placeholder": "Brno",
      "prefixCls": "ant-select",
      "showSearch": false,
      "transitionName": "slide-up",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "choiceTransitionName": "zoom",
        "className": "input",
        "dataSource": Array [],
        "filterOption": false,
        "onChange": [Function],
        "optionLabelProp": "children",
        "placeholder": "Brno",
        "prefixCls": "ant-select",
        "showSearch": false,
        "transitionName": "slide-up",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
